zimui/src/components/TopicSection.vue

```diff
@@ -11,33 +14,59 @@ defineProps({
 const instance = getCurrentInstance()
 const uid = ref('carroussel_' + instance?.uid)
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-const splitChunks = (inputArray: any[], perChunk: number) => {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  return inputArray.reduce((all: any[], one: ConcatArray<never>, i: number) => {
-    const ch = Math.floor(i / perChunk)
-    all[ch] = [].concat(all[ch] || [], one)
-    return all
-  }, [])
-}
-
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-const limitCardsPerSections = (inputArray: any[], section_slug: string) => {
+/**
+ * Keep only the 10 first items of the input array. If more than 10 items
+ * where present in the list, a "magic/virtual" 11th item is created to indicate
+ * that more items are available.
+ * @param subsections - array of items to limit
+ * @param sectionSlug - amount of items per chunk
+ */
+const limitCardsPerSections = (
+  subsections: TopicSubSection[],
+  sectionSlug: string,
+): TopicCardData[] => {
   const maxCardPerSection = 10
-  if (inputArray.length > maxCardPerSection) {
-    const inputSliced = inputArray.slice(0, maxCardPerSection)
-    inputSliced.push({
+
+  if (subsections.length > maxCardPerSection) {
+    const slicedInput = subsections
+      .slice(0, maxCardPerSection)
+      .map(transformTopicSectionOrSubSectionToCardData)
+    slicedInput.push({
       kind: 'more',
-      count_more: inputArray.length - maxCardPerSection,
-      slug: section_slug,
-    })
-    return inputSliced
+      count_more: subsections.length - maxCardPerSection,
+      slug: sectionSlug,
+    } as TopicCardData)
+    return slicedInput
   }
-  return inputArray
+  return subsections.map(transformTopicSectionOrSubSectionToCardData)
+}
+
+/**
+ * Split an array of cards into chunks of cards. Each chunk is an array
+ * of cards, with at most perChunk cards.
+ * @param cards - array of cards to split into chunks
+ * @param cardsPerChunk - amount of cards per chunk
+ */
+const splitCardsListIntoChunks = (
+  cards: TopicCardData[],
+  cardsPerChunk: number,
+): TopicCardData[][] => {
+  return cards.reduce(
+    (all: TopicCardData[][], one: TopicCardData, i: number) => {
+      const ch = Math.floor(i / cardsPerChunk)
```

Comment:
I think we can afford words for variable name: `chunk`, `index`. Even `all` is vague

Reply:
Done